### ISSUE TITLE
Access logs on server side encrypted bucket?

### ISSUE THREAD

**Issue by durnote** [terraform-aws-modules/terraform-aws-alb#61]
Because of standards I need to make sure my ALB access logs are also encrypted.
Is this supported ? I am using standard S3 KMS as default encryption for S3 bucket .

When I try to do it I am getting this error:

aws_alb.my_alb: Failure configuring LB attributes: InvalidConfigurationRequest: Access Denied for bucket: my.bucketname. Please check S3bucket permission

**Comment by vas1468** [comment]
There is no such option documented/supported on terraform aws_lb resource either.
[https://www.terraform.io/docs/providers/aws/r/lb.html](url)
[https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html](url)

**Comment by durnote** [comment]
Thanks @brandoconnor for links and thoughts. Yep, I have that very special policy (the one that allows ALB's account id to access my S3) it is exact the same as the one generated using your scripts ( I tried both). However, I am still facing the same error - but only when I use encrypted bucket.  With non-encrypted bucket works fine.  

This is how policy looks like on my end  (I had stripped some sensitive ids only) :
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "AllowToPutLoadBalancerLogsToS3Bucket",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::AWS-ACCOUNTID-FOR_US-EAST-1-GOES-HERE:root"
            },
            "Action": "s3:PutObject",
            "Resource": "arn:aws:s3:::logs.mybucketname/logs-alb/AWSLogs/MY-ACCOUNTID-GOES-HERE/*"
        }
    ]
}
```
For AWS docs, yeah, I was also confused with that part and read it twice. It looks like they do automatically encrypting.  Listing S3 objects with boto shows me that alb log objects are really 

`Encryption:AES256`

However, they don't mention if using buckets with AWS-KMS is supported or not.  I guess it would be nice to know that also.  Any ways, hope this helps someone who comes comes to the same question :)
